resolver/master_changer_test.go

```diff
@@ -0,0 +1,91 @@
+package resolver
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+)
+
+func assertMastersState(mc *MasterChanger, want []string) error {
+	var got []string
+
+	select {
+	case got = <-mc.ChangedChan():
+	default:
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		return fmt.Errorf("got %q, want %q", got, want)
+	}
+
+	return nil
+}
+
+func TestMasterChanger_UpdatedMasters(t *testing.T) {
+	mc := NewMasterChanger([]string{"b:5050", "a:5050"}, 1)
+
+	for i, tt := range []struct {
+		masters, want []string
+	}{
+		{
+			[]string{"c", "d", "b"},
+			[]string{"", "b:5050", "c:5050", "d:5050"},
```

Comment:
As someone who just drops by this test, it's surprising to see that we expect an empty string as the first value. This needs more documentation and explanation.